### PR TITLE
Allow old prepped RFCs to be rendered

### PR DESCRIPTION
**Pull request by jennifer-richards**
This addresses [ticket 664](https://trac.ietf.org/trac/xml2rfc/ticket/664).

Prepped XML files should already have pn attributes for all the relevant elements. These will match any references in xref elements that were, e.g., in a precomputed index or table of contents. When we changed the format of the pns we generate, this broke those precomputed xrefs.

This leaves any incoming pn values alone. These should ordinarily be present for every element on a prepped XML, so warnings are emitted if one is missing, and the value we would have assigned is given to it. (This may result in bad linking in some pathological cases, but the input really is not valid if this arises - and the user got a warning.) When passing through an existing pn, a warning is also emitted if the old value is different from the current value, as a hint that the prepped XML is out of date.

The HTML writer is updated to catch both old and new appendix pns when generating IDs for section elements.

This also fixes a corner case in the author list handling that broke several existing rfc documents.

You will note that there are no new tests. I plan to add one of the RFCs that was failing, but after beating my head against the tests for a few hours already, I still am not able to run them. When I get that sorted out, I will add a prepped rfc to tests/input and its expected output to the tests/valid and add it to the list run by the Makefile.

### CODE REVIEW

**Comment by jennifer-richards** [comment]
One other comment: my actual patch is against 3.9.1, the current release, but I have not yet updated our git repo to have that. The relevant code in dev/3.8.0 is unchanged in the new release, so I don't want to hold this up on the book keeping.

**Comment by jennifer-richards** [comment]
And one other other comment: while I have not been able to run the real tests, I've run all the prepped RFCs since around 8600 through this and they seem to do the right thing.

File: cli/xml2rfc/writers/base.py [hunk 1]
```diff
@@ -2250,7 +2250,8 @@ def level_of_section_num(num):
     def is_top_level_section(cls, num):
         return cls.level_of_section_num(num) == 1
 
-    @staticmethod
-    def is_appendix(pn):
+    appendix_pn_re = re.compile(r'^section-[a-z]\.|^section-appendix\.')
```
Comment (meadmaker) [inline cli/xml2rfc/writers/base.py:2253]: I know this is absurd, but what happens if a document has twenty-seven appendices?   What comes after `section-z`?

File: cli/xml2rfc/writers/preptool.py [hunk 2]
```diff
@@ -1282,18 +1291,18 @@ def references_check_usage(self, e, p):
...
+        self.set_element_pn(e, '%s-toc.%s' % (pnprefix[e.tag], self.toc_section_number, ))
 
     def front_abstract_add_number(self, e, p):
-        e.set('pn', '%s-abstract' % pnprefix[e.tag])
+        self.set_element_pn(e, '%s-abstract' % pnprefix[e.tag])
 
     def front_note_add_number(self, e, p):
         self.note_number += 1
-        e.set('pn', '%s-note.%s' % (pnprefix[e.tag], self.note_number, ))
+        self.set_element_pn(e, '%s-note.%s' % (pnprefix[e.tag], self.note_number, ))
```
Comment (meadmaker) [inline cli/xml2rfc/writers/preptool.py:1308]: Is it worth noting that this tag doesn't match the comment above on line 1282?

**Review by meadmaker — APPROVED** [review]
Generally fine.  Commit it when you feel comfortable.

File: cli/xml2rfc/writers/base.py (hunk 1 above)
Comment (jennifer-richards) [reply cli/xml2rfc/writers/base.py:2253]: The system throws an error if there  are too  many:
https://github.com/painless-security/xml2rfc/blob/6041cc1d4b5596d96cf4618826ee5df1c6075c43/cli/xml2rfc/writers/preptool.py#L1347-L1350

File: cli/xml2rfc/writers/preptool.py (hunk 2 above)
Comment (jennifer-richards) [reply cli/xml2rfc/writers/preptool.py:1308]: Probably a good idea. Those are from an old document that disagrees in more than a few places.
